packages/advanced-logic/specs/payment-network-erc20-fee-proxy-contract-0.1.0.md

```diff
@@ -0,0 +1,328 @@
+| ------------------------- | -------------------------------------------------------------- |
+| **id**                    | "pn-erc20-fee-proxy-contract"                                  |
+| **type**                  | "paymentNetwork"                                               |
+| **version**               | "0.1.0"                                                        |
+| **values**                |                                                                |
+| **values.paymentAddress** | `paymentAddress` from parameters if given, undefined otherwise |
+| **values.refundAddress**  | `refundAddress` from parameters if given, undefined otherwise  |
+| **values.feeAddress**     | `feeAddress` from parameters if given, undefined otherwise     |
+| **values.feeAmount**      | `feeAmount` from parameters if given, undefined otherwise      |
+| **values.salt**           | Salt for the request                                           |
+| **events**                | Array with one 'create' event (see below)                      |
+
+the 'create' event:
+
+|  Property                     |  Value                                                         |
+| ----------------------------- | -------------------------------------------------------------- |
+| **name**                      | 'create'                                                       |
+| **parameters**                |                                                                |
+| **parameters.paymentAddress** | `paymentAddress` from parameters if given, undefined otherwise |
+| **parameters.refundAddress**  | `refundAddress` from parameters if given, undefined otherwise  |
+| **parameters.feeAddress**     | `feeAddress` from parameters if given, undefined otherwise     |
+| **parameters.feeAmount**      | `feeAmount` from parameters if given, undefined otherwise      |
+| **parameters.salt**           | Salt for the request                                           |
+
+---
+
+### Updates
+
+#### addPaymentAddress
+
+##### Parameters
+
+|                               | Type   | Description                                   | Requirement   |
+| ----------------------------- | ------ | --------------------------------------------- | ------------- |
+| **id**                        | String | Constant value: "pn-erc20-fee-proxy-contract" | **Mandatory** |
+| **action**                    | String | Constant value: "addPaymentAddress"           | **Mandatory** |
+| **parameters**                | Object |                                               |               |
+| **parameters.paymentAddress** | String | Ethereum address for the payment              | **Mandatory** |
+
+##### Conditions
+
+This action is valid, if:
+
+- The extension state with the id "pn-erc20-fee-proxy-contract" exists
+- The signer is the `payee`
+- The extension property `paymentAddress` is undefined
+
+##### Warnings
+
+None.
```

Comment:
Shouldn't there be a warning if the payer tries to add a payment address?

Reply:
Not necessarilly, because this case will never pass the conditions above.

Reply:
To clarify this and the following points about the warnings:
We will show warnings only when a request follow the conditions, but is still something to worry about. 
For example, when the payer creates the request and sets the feeAddress, it can be a valid use-case, but it can be a scam. So we allow that usecase, but we warn the builder that this can be something sketchy.